src/main/java/org/mockito/internal/util/eventbus/EventBus.java

```diff
@@ -0,0 +1,82 @@
+package org.mockito.internal.util.eventbus;
+
+import static org.mockito.internal.util.Checks.checkNotNull;
+import static org.mockito.internal.util.eventbus.compability.ListenerSupport.legacyListenerOf;
+
+import java.lang.reflect.Method;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.Set;
+import org.mockito.listeners.Subscribe;
+
+public class EventBus {
+
+    private final Set<Subscriber> eventHandlers = new HashSet<Subscriber>();
+
+    public void register(Object listener) {
+        checkNotNull(listener, "The listener must not be null!");
+
+        eventHandlers.addAll(eventHandlerOf(listener));
+        eventHandlers.addAll( legacyListenerOf(listener));
+    }
+
+    
+
+    public void post(Object event) {
+        checkNotNull(event, "The event must not be null!");
+
+        for (Subscriber eventHandler : eventHandlers) {
+            eventHandler.tryHandleEvent(event);
+        }
+    }
+    
+    public void unregister(Object listener) {
```

Comment:
Actually no, `eventHandlers` is a set of Subscriber instances, every Subscriber  represents a listener method of a listener instance. So every subscriber of the listener instance must be removed. The name `eventHandlers` is confusing.

If the mockito team decides to introduce an EventBus I will start more refactoring and polishing.